File: LSGi/source/queryService/clientService/QueryClient.cpp
/**
“© Copyright 2017  Hewlett Packard Enterprise Development LP

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.”
*/


//============================================================================
// Name        : QueryClient.cpp
// Author      : gomariat
// Version     :
// Copyright   : Your copyright notice
// Description : Hello World in C, Ansi-style
//============================================================================

#include <iostream>
#include <stdio.h>
#include <stdlib.h>
#include "clientSocket.h"


#define LOCALHOST

int SERVICEPORT=58000;
/*server to connect*/

void runQueryClient(queryRequest & request){

	std::cout <<"\n -- Getting inference state: connecting to port=" << SERVICEPORT<<"--";
	connectToServer(SERVICEPORT, "localhost",request);
}
/*
 * Main Driver for the query service client
 * This client will connect a client to request
 * the type of query over the inference.
 *
 * Willl have 3 types of request:
 *
 * 1.search state of a given vertex id
 * 2.retrieve the state of all the vertices
 * 3.retrive the state of a vertices that P(v)>=threshold.
 *
 * */
int main(int argc, char* argv[]) {


	if(argc<3){
		std::cout <<"\nIncorrect Usage: <type of query=1 (byVId), =3 (byThreshold3> <value> <port =default 58000)\n";
		std::cout <<"\nExample: ./clientService 1 0 -> for query type 1 ";
		return 0;
	}

	queryRequest request;
	request.type=atoi(argv[1]);
	request.value=atoi(argv[2]);
	if (argc ==4 ){
		SERVICEPORT=atoi(argv[3]);
	}

	/*send query to inference service*/
	runQueryClient(request);

	std::cout <<std::endl;
	return EXIT_SUCCESS;
}

File: LSGi/source/webConsole/JClient/src/main/java/com/labs/hpe/service/QueryClient.java
/**
 * 
 */
package com.labs.hpe.service;

//import org.apache.log4j.Logger;

import com.labs.hpe.model.QueryRequest;

/**
 * This client will connect a client to request 
 * the type of query over the inference.
 * @author Janneth Rivera, Tere Gonzalez
 * Feb 23, 2016
 */
public class QueryClient {

	//private static final Logger logger = Logger.getLogger(QueryClient.class);
	
	private String serverAddress="localhost";//"192.168.122.172"; 
	private int serverPort=58000;
	
	public void runQueryClient(QueryRequest request){
		try{
			System.out.println("-- Getting inference state: connecting to " + serverAddress + ":" + serverPort + "--");
			ClientSocket cs = new ClientSocket();
			//cs.connectToServer(serverPort, serverAddress, request);
			cs.connect(serverPort, serverAddress);
			cs.send(request);
			cs.receive();			
			cs.close();
		}catch(Exception e){        	
        	//logger.error(e);
			e.printStackTrace();
        }
	}
	


	/**
	 * @return the serverAddress
	 */
	public String getServerAddress() {
		return serverAddress;
	}



	/**
	 * @param serverAddress the serverAddress to set
	 */
	public void setServerAddress(String serverAddress) {
		this.serverAddress = serverAddress;
	}



	/**
	 * @return the serverPort
	 */
	public int getServerPort() {
		return serverPort;
	}



	/**
	 * @param serverPort the serverPort to set
	 */
	public void setServerPort(int serverPort) {
		this.serverPort = serverPort;
	}



	/**
	 * Main Driver for the query service client
	 * This client will connect a client to request
	 * the type of query over the inference.
	 *
	 * Will have 3 types of request:
	 *
	 * 1.search state of a given vertex id
	 * 2.retrieve the state of all the vertices
	 * 3.Retrieve the state of a vertices that P(v)>=threshold.
	 *
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		try{
			if(args.length < 4){
				System.out.println("Incorrect Usage: <type of query=1 (byVId), =3 (byThreshold3), =5 (byTresholdStats) <value> <server =default localhost> <port =default 58000> )");
				System.out.println("Example: ./clientService 1 0 -> for query type 1 ");
				return;
			}
			
				
			QueryRequest request = new QueryRequest();
			request.type = Integer.parseInt(args[0]);
			request.value = Integer.parseInt(args[1]);		
			
			/*send query to inference service*/		
			QueryClient qc = new QueryClient();
			qc.setServerAddress(args[2]);
			qc.setServerPort(Integer.parseInt(args[3]));
			qc.runQueryClient(request);
			
		}catch(Exception e){
        	//logger.error(e);
        }
	}

}
